src/endpoints/accounts/account.service.ts

```diff
@@ -345,4 +348,46 @@ export class AccountService {
 
     return await this.elasticService.getCount('scdeploys', elasticQuery);
   }
+
+
+  async getAccountHistory(address: string, pagination: QueryPagination): Promise<AccountHistory[]> {
+    const elasticQuery: ElasticQuery = AccountService.buildAccountHistoryFilterQuery(address);
+    elasticQuery
+        .withPagination(pagination)
+        .withSort([{ name: 'timestamp', order: ElasticSortOrder.descending }]);
+
+    const elasticResult = await this.elasticService.getList('accountshistory', 'address', elasticQuery);
+    return elasticResult.map(item => ApiUtils.mergeObjects(new AccountHistory(), item));
+  }
+
+  private static buildAccountHistoryFilterQuery(address?: string, token?: string): ElasticQuery {
+    const shouldQueries: AbstractQuery[] = [];
+    const mustQueries: AbstractQuery[] = [];
+
+    if (address) {
+      mustQueries.push(QueryType.Match('address', address));
+    }
+
+    if (token) {
+      mustQueries.push(QueryType.Match('token', token));
+    }
+
+    return ElasticQuery.create()
+      .withCondition(QueryConditionOptions.should, shouldQueries)
+      .withCondition(QueryConditionOptions.must, mustQueries);
+  }
+
+  async getAccountTokenHistory(address: string, tokenIdentifier: string, pagination: QueryPagination): Promise<AccountEsdtHistory[]> {
+    if(!tokenIdentifier){
```

Comment:
The tokenIdentifier can't be undefined or null so this check will have no effect.